elastic/kibana
# [Lens] Add styling options for x and y axes on the settings popover

## Summary

Closes #68314 and #68313. Adds styling options for both x and y axes.

This PR adds more settings to the settings popover.
<img width="396" alt="Screenshot 2020-07-21 at 1 12 01 PM" src="https://user-images.githubusercontent.com/17003240/88041933-cbcaa180-cb53-11ea-942d-cc24bcc06a22.png">

By default, the tick labels for x and y axes are enabled ~~while the gridlines ones are disabled~~.
**Update** : We decided to go with the gridlines enabled by default

User can also hide the titles of x and y axes by switching off the corresponding switch. If the switch is on, then the user can also overwrite the title of each axis.
<img width="1395" alt="Screenshot 2020-07-21 at 1 13 57 PM" src="https://user-images.githubusercontent.com/17003240/88042090-0df3e300-cb54-11ea-94f6-681c02399c47.png">

Note: I assume that if the user has both left and right y axes, then the y-axis title setting overwrites both. For this reason, only one y-axis title appears on the screenshot above. If we want to give the possibility to the user to change both titles then it needs an extra setting and makes things a little bit more complicated.

### Checklist

Delete any items that are not applicable to this PR.

- [x] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers) - Safari, FF and Chrome

### What to test

1. Check that your previous Lens Charts render correctly
2. Create a new Chart, check that the defaults are correct, change the settings and check that everything works properly
3. Add extra layers
4. Check the suggestions
5. Add lens charts to a dashboard and check that everything works fine
....
